sd-2019-stack-overflow-frontend/src/view/SmartQuestionList.js

```diff
@@ -0,0 +1,124 @@
+import React, {Component} from "react";
+import sOUserModel from "../model/sOUserModel";
+import questionModel from "../model/questionModel";
+import questionListPresenter from "../presenter/questionListPresenter";
+import navbarPresenter from "../presenter/navbarPresenter";
+import QuestionList from "./QuestionList";
+import NavbarNotLoggedIn from "./NavbarNotLoggedIn";
+import NavbarLoggedIn from "./NavbarLoggedIn";
+import SignUpModal from "./SignUpModal";
+import LogInModal from "./LogInModal";
+import QuestionOperations from "./QuestionOperations";
+import AskQuestionModal from "./AskQuestionModal";
+import Footer from "./Footer";
+
+const mapModelStateToComponentState = (questionModelState, sOUserModelState) => (
+    {
+        questions: questionModelState.questions,
+        askQuestionModalActive: questionModelState.askQuestionModalActive,
+        newQuestion: questionModelState.newQuestion,
+        newTag: questionModelState.newTag,
+        titleFilter: questionModelState.titleFilter,
+        tagFilter: questionModelState.tagFilter,
+        loggedInUsername: sOUserModelState.loggedInUsername,
+        signUpModalActive: sOUserModelState.signUpModalActive,
+        newSOUserUsername: sOUserModelState.newSOUser.username,
+        newSOUserPassword: sOUserModelState.newSOUser.password,
+        logInModalActive: sOUserModelState.logInModalActive,
+        logInSOUserUsername: sOUserModelState.logInSOUser.username,
+        logInSOUserPassword: sOUserModelState.logInSOUser.password
+    }
+);
+
+export default class SmartQuestionList extends Component {
```

Comment:
This is a HUGE smart component... I would split it up into several smaller ones.